src/ring_theory/graded_algebra/radical.lean

```diff
@@ -0,0 +1,332 @@
+    simp only [not_and, prod.mk.inj_iff, ne.def, dfinsupp.mem_support_to_fun, finset.mem_singleton,
+      finset.mem_product] at Hij,
+    exact Hij.1.2 Hij.2.1 Hij.2.2, },
+end⟩
+
+lemma homogeneous_ideal.rad_eq (I : homogeneous_ideal A) :
+  I.1.radical = Inf {J | I.1 ≤ J ∧ ideal.is_homogeneous A J ∧ J.is_prime} :=
+begin
+  have subset₁ : I.1.radical ≤ Inf {J | I.1 ≤ J ∧ ideal.is_homogeneous A J ∧ J.is_prime},
+  { rw ideal.radical_eq_Inf, intros x hx,
+    rw [submodule.mem_Inf] at hx ⊢, intros J HJ, apply hx,
+    obtain ⟨HJ₁, _, HJ₂⟩ := HJ,
+    refine ⟨HJ₁, HJ₂⟩, },
+  have subset₂ : Inf {J | I.1 ≤ J ∧ ideal.is_homogeneous A J ∧ J.is_prime} ≤ I.1.radical,
+  { intros x hx,
+    rw ideal.radical_eq_Inf,
+    rw [submodule.mem_Inf] at hx ⊢,
+    rintros J ⟨HJ₁, HJ₂⟩,
+    specialize hx (ideal.homogeneous_core A J) _,
+    refine ⟨_, (ideal.homogeneous_core A _).2, _⟩,
+    { have HI := I.2,
+      rw [ideal.is_homogeneous.iff_eq] at HI,
+      rw ← HI,
+      apply ideal.span_mono, intros y hy,
+      obtain ⟨z, ⟨hz₁, rfl⟩⟩ := hy,
+      rw set.mem_preimage at hz₁,
+      specialize HJ₁ hz₁,
+      refine ⟨z, _, rfl⟩,
+      simpa [set.mem_preimage] using HJ₁, },
+    { set J' := ideal.homogeneous_core A J with eq_J',
+      have homogeneity₀ := (ideal.homogeneous_core A J).2,
+      apply homogeneous_ideal.is_prime_iff A ⟨J', homogeneity₀⟩,
+      intro rid,
+      have rid' : J = ⊤,
+      { have : J'.1 ≤ J := ideal.coe_homogeneous_core_le A J,
+        simp only [homogeneous_ideal.eq_top_iff] at rid,
+        erw ← subtype.val_eq_coe at rid,
+        erw rid at this,
+        rw top_le_iff at this,
+        exact this, },
+      apply HJ₂.1, exact rid',
+      rintros x y hx hy hxy,
+      have H := HJ₂.mem_or_mem (ideal.coe_homogeneous_core_le A J hxy),
+      cases H,
+      { left,
+        have : ∀ i : ι, (graded_algebra.decompose A x i : R) ∈
+          (⟨J', homogeneity₀⟩ : homogeneous_ideal A),
+        { intros i, apply homogeneity₀, apply ideal.subset_span,
+          simp only [set.mem_inter_eq, set_like.mem_coe, set.mem_set_of_eq],
+          refine ⟨⟨x, hx⟩, H, rfl⟩, },
+        rw ←graded_algebra.sum_support_decompose A x,
+        apply ideal.sum_mem J'.1,
+        intros j hj,
+        apply this, },
+      { right,
+        have : ∀ i : ι, (graded_algebra.decompose A y i : R) ∈
+          (⟨J', homogeneity₀⟩ : homogeneous_ideal A),
+        { intros i, apply homogeneity₀, apply ideal.subset_span,
+          simp only [set.mem_inter_eq, set_like.mem_coe, set.mem_set_of_eq],
+          refine ⟨⟨y, hy⟩, H, rfl⟩, },
+        rw ←graded_algebra.sum_support_decompose A y,
+        apply ideal.sum_mem J'.1,
+        intros j hj, apply this, }, },
+      refine (ideal.coe_homogeneous_core_le A J) hx, },
+
+  ext x, split;
+  intro hx,
+  { exact subset₁ hx },
+  { exact subset₂ hx },
+end
+
+lemma homogeneous_ideal.rad (I : homogeneous_ideal A)  :
+  ideal.is_homogeneous A I.1.radical :=
```

Comment:
This statement should be
```lean
lemma is_homogeneous.radical {I : ideal R} (h : I.is_homogeneous A) :
  I.radical.is_homogeneous A:=
```
I imagine the proof will have to change a bit, but not in a meaningful way.

Reply:
surprisingly, the proof didn't need to change!

Reply:
The one you pushed isn't the same as the one in my comment

Reply:
sorry, it I fixed now